src/tracing_tool_output.ml

```diff
@@ -153,34 +154,60 @@ let maybe_stash_old_trace ~filename =
   | Core_unix.Unix_error (ENOENT, (_ : string), (_ : string)) -> ()
 ;;
 
-let write_and_maybe_serve ?num_temp_strs t ~filename ~f_sexp ~f_fuchsia =
+let write_and_maybe_serve
+  ?num_temp_strs
+  t
+  ~filename
+  ~(f :
+     ?events_writer:events_writer
+     -> ?writer:Tracing_zero.Writer.t
+     -> unit
+     -> 'a Deferred.Or_error.t)
+  =
   let open Deferred.Or_error.Let_syntax in
   maybe_stash_old_trace ~filename;
-  match t.output with
-  | `Sexp store_path -> Writer.with_file_atomic store_path ~f:f_sexp
-  | `Fuchsia store_path ->
-    let fd = Core_unix.openfile store_path ~mode:[ O_RDWR; O_CREAT; O_CLOEXEC ] in
-    (* Write to and serve from an indirect reference to [store_path], through our process'
-     fd table. This is a little grotesque, but avoids a race where the user runs
-     magic-trace twice with the same [-output], such that the filename changes under
-     [-serve] -- without this hack, the earlier magic-trace serving instance would start
-     serving the new trace, which is unlikely to be what the user expected. *)
+  let { serve; output_path } = t in
+  match String.is_suffix ~suffix:".fxt" output_path with
```

Comment:
Is there a naming convention for Fuchsia format files and for Binio files? I've seen both `.ftf` and `.fxt` used for Fuchsia. This PR requires it to be `.fxt` and gives an error if the output file name extension is not one `.sexp .fxt .binio`, whereas before there was no such restriction. Also, should it be ".bin" instead of ".binio"?

Reply:
`.fxt` is the only file extension for fuchsia trace files. `.ftf` was present in pre-release magic-trace because we didn't know what the right file extension was.

I prefer `.binio` to `.bin`, but I don't know if there's a standard naming convention. I prefer `.binio` because people sometimes use `.bin` to refer to other things, whereas I've never seen `.binio` used for anything else.

Reply:
So do we think it's fine the way it is now or should I change something? For example, assume fuchsia by default and only 
change to binio or sexp if extension matches

Reply:
I like "assume fuchsia by default"